image/docker_schema2.go

```diff
@@ -173,6 +168,50 @@ func (m *manifestSchema2) UpdatedImage(options types.ManifestUpdateOptions) (typ
 	return memoryImageFromManifest(&copy), nil
 }
 
+// Edits the layers in the manifest when called by replacing them with the
+// appropriate infos provided. This is metadata-only -- the actual layers still
+// have to get to the image.
+func (m *manifestSchema2) performEdits(infos []types.BlobInfo) error {
+	infolen := len(infos)
+
+	if infolen < 1 {
+		return errors.New("cannot edit an image down to less than one layer")
+	}
+
+	// regurgitate the image configuration for recalculation of layers in the event the layer list has been edited.
+	imageConfig := &image{}
+	blob, err := m.ConfigBlob()
+	if err != nil {
+		return err
+	}
+
+	if err := json.Unmarshal(blob, imageConfig); err != nil {
+		return err
+	}
+
+	imageConfig.History = make([]imageHistory, infolen)
+	imageConfig.RootFS.DiffIDs = make([]digest.Digest, infolen)
+	m.LayersDescriptors = make([]descriptor, infolen)
+	for i, info := range infos {
+		imageConfig.History[i] = imageHistory{}
+		imageConfig.RootFS.DiffIDs[i] = info.Digest
+		m.LayersDescriptors[i].Digest = info.Digest
+		m.LayersDescriptors[i].Size = info.Size
+		m.LayersDescriptors[i].URLs = info.URLs
+	}
+	imageConfig.RootFS.BaseLayer = m.LayersDescriptors[0].Digest.String()
```

Comment:
(Note to self: I didn’t review the `config.json` edits in detail here, … but this _is_ inconsistent with what `manifestSchema1.convertToManifestSchema2` does.)